doc/release/1.8.0-notes.rst

```diff
@@ -26,104 +26,876 @@ For running on PyPy, PyPy3 6.0+ is required.
+==============================
+
+`scipy.integrate.quad_vec` introduces a new optional keyword-only argument,
+``args``. ``args`` takes in a tuple of extra arguments if any (default is
+``args=()``), which is then internally used to pass into the callable function
+(needing these extra arguments) which we wish to integrate.
 
 `scipy.interpolate` improvements
 ================================
 
+`scipy.interpolate.BSpline` has a new method, ``design_matrix``, which
+constructs a design matrix of b-splines in the sparse CSR format.
+
+A new method ``from_cubic`` in ``BSpline`` class allows to convert a
+``CubicSpline`` object to ``BSpline`` object.
 
 `scipy.linalg` improvements
 ===========================
 
-
-`scipy.ndimage` improvements
-============================
-
+`scipy.linalg` gained three new public array structure investigation functions.
+`scipy.linalg.bandwidth` returns information about the bandedness of an array
+and can be used to test for triangular structure discovery, while
+`scipy.linalg.issymmetric` and `scipy.linalg.ishermitian` test the array for
+exact and approximate symmetric/Hermitian structure.
 
 `scipy.optimize` improvements
 =============================
 
+`scipy.optimize.check_grad` introduces two new optional keyword only arguments,
+``direction`` and ``seed``. ``direction`` can take values, ``'all'`` (default),
+in which case all the one hot direction vectors will be used for verifying
+the input analytical gradient function and ``'random'``, in which case a
+random direction vector will be used for the same purpose. ``seed``
+(default is ``None``) can be used for reproducing the return value of
+``check_grad`` function. It will be used only when ``direction='random'``.
+
+The `scipy.optimize.minimize` ``TNC`` method has been rewritten to use Cython
+bindings. This also fixes an issue with the callback altering the state of the
+optimization.
+
+Added optional parameters ``target_accept_rate`` and ``stepwise_factor`` for
+adapative step size adjustment in ``basinhopping``.
+
+The ``epsilon`` argument to ``approx_fprime`` is now optional so that it may
+have a default value consistent with most other functions in `scipy.optimize`.
 
 `scipy.signal` improvements
 ===========================
 
+Add ``analog`` argument, default ``False``, to ``zpk2sos``, and add new pairing
+option ``'minimal'`` to construct analog and minimal discrete SOS arrays.
+``tf2sos`` uses zpk2sos; add ``analog`` argument here as well, and pass it on
+to ``zpk2sos``.
+
+``savgol_coeffs`` and ``savgol_filter`` now work for even window lengths.
+
+Added the Chirp Z-transform and Zoom FFT available as `scipy.signal.CZT` and
+`scipy.signal.ZoomFFT`.
 
 `scipy.sparse` improvements
 ===========================
 
+An array API has been added for early testing and feedback. This work is
+ongoing, and users may expect backwards compatibility changes in
+future releases. Please refer to the `scipy.sparse`
+docstring for more information.
```

Comment:
My original text here was:

> An array API has been added to facilitate the ecosystem-wide transition away from the matrix API. We recommend that new work uses these array objects: bsr_array, coo_array, etc. Please refer to the scipy.sparse docstring for more information.

OK with not recommending the array objects explicitly in the release notes, but as above I am confused by the "expect backwards compatibility changes" segment.

Reply:
Agreed, that seems incorrect. What's there should be solid.

Reply:
It was based on a suggestion from @ev-br above: https://github.com/scipy/scipy/pull/15161#discussion_r762903968

> How about something along the lines of "This work is ongoing. We would appreciate a wide testing of this new functionality, but users may expect backwards compatibility issues in next releases."

I'm not sure what is incorrect or confusing about it--I think Evgeni and I are simply suggesting that it could be subject to change in future releases, which would be a departure from the usual rock-solid backwards compatiblity guarantees you usually get in SciPy.

Reply:
I think because "backward-compatibility changes" is a confusing term; "backward-incompatible changes" makes sense, but users don't have to worry about "backward-compatible changes".  The only item I am aware of that will generate incompatible changes is once we have a sparse-1d-array object.

Perhaps:

"This work is ongoing, so we appreciate thorough testing of the new array classes. Users should expect minor API refinements over the next few releases."